src/test/java/api/loans/CheckInByBarcodeTests.java

```diff
@@ -1611,6 +1612,72 @@ void canCheckinItemWhenRequestForAnotherItemOfSameInstanceExists() {
       allOf(isOpenNotYetFilled(), hasPosition(1)));
   }
 
+  @Test
+  void canFulFillRecallRequestWhenCheckInAnotherItemOfSameInstance() {
+    configurationsFixture.enableTlrFeature();
+    List<ItemResource> items = itemsFixture.createMultipleItemsForTheSameInstance(2);
+    ItemResource firstItem = items.get(0);
+    ItemResource secondItem = items.get(1);
+
+    checkOutFixture.checkOutByBarcode(firstItem, usersFixture.jessica());
+    checkOutFixture.checkOutByBarcode(secondItem, usersFixture.james());
+
+    IndividualResource recallRequest = requestsFixture.placeTitleLevelRecallRequest(
+      firstItem.getInstanceId(), usersFixture.steve());
+    assertThat(recallRequest.getJson(), allOf(isOpenNotYetFilled(), hasPosition(1)));
+    assertThat(recallRequest.getJson().getString("itemId"), is(firstItem.getId().toString()));
+
+    CheckInByBarcodeResponse secondItemCheckInResponse = checkInFixture.checkInByBarcode(secondItem);
+    assertThat(secondItemCheckInResponse.getItem(), isAwaitingPickup());
+
+    JsonObject recallRequestResponse = requestsFixture.getById(recallRequest.getId()).getJson();
+    assertThat(recallRequestResponse, allOf(isOpenAwaitingPickup(), hasPosition(1)));
+    assertThat(recallRequestResponse.getString("itemId"), is(secondItem.getId().toString()));
+
+    CheckInByBarcodeResponse firstItemCheckInResponse = checkInFixture.checkInByBarcode(firstItem);
+    assertThat(firstItemCheckInResponse.getItem(), isAvailable());
+  }
+
+  @Test
+  void canFulFillRecallRequestWhenCheckInAnotherItemOfSameInstanceWithMultipleRecallRequests() {
+    configurationsFixture.enableTlrFeature();
+    List<ItemResource> items = itemsFixture.createMultipleItemsForTheSameInstance(3);
+    ItemResource firstItem = items.get(0);
+    ItemResource secondItem = items.get(1);
+    ItemResource thirdItem = items.get(2);
+
+    checkOutFixture.checkOutByBarcode(firstItem, usersFixture.jessica());
+    checkOutFixture.checkOutByBarcode(thirdItem, usersFixture.james());
+    checkOutFixture.checkOutByBarcode(secondItem, usersFixture.rebecca());
+
+    IndividualResource recallRequest1 = requestsFixture.placeTitleLevelRecallRequest(
+      firstItem.getInstanceId(), usersFixture.steve());
```

Comment:
Please extract `instanceId` into a variable and use it everywhere, so that it's clear that all three items belong to the same instance.

Reply:
extracted, thanks